### PR TITLE
Fix breadcrumb and formatting issues

### PR DESCRIPTION
**Pull request by FinalAngel**
- fixes url references
- fixes breadcrumb in ``admin/cms/page/recover/{id}/`` to reflect "django CMS" as name
- fixes codestyle

### CODE REVIEW

File: cms/templates/admin/cms/page/history/recover_header.html [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-    <a href="../../../">{{ app_label|capfirst|escape }}</a> &rsaquo;
-    <a href="../../">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
-    <a href="../">{% blocktrans with opts.verbose_name as verbose_name %}Recover deleted {{ verbose_name }}{% endblocktrans %}</a> &rsaquo;
-    {{ title }}
-</div>
+    <div class="breadcrumbs">
+        <a href="{% url 'admin:index' %}">{% trans "Home" %}</a> &rsaquo;
+        <a href="{% url 'admin:app_list' app_label=opts.app_label %}">{{ opts.app_config.verbose_name }}</a> &rsaquo;
+        <a href="{% url opts|admin_urlname:'changelist' %}">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
+        <a href="../">
```
Comment (FinalAngel) [inline cms/templates/admin/cms/page/history/recover_header.html]: @yakky @mkoistinen is there a way that I can also replace this with a {% url %}? I couldn't find the match

File: cms/templates/admin/cms/page/history/revision_header.html [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
-    <a href="../../../">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
-    <a href="../../">{{ original|truncatewords:"18" }}</a> &rsaquo;
-    <a href="../">{% trans "History" %}</a> &rsaquo;
-    {% blocktrans with opts.verbose_name as verbose_name %}Revert {{ verbose_name }}{% endblocktrans %}
-</div>
+    <div class="breadcrumbs">
+        <a href="{% url 'admin:index' %}">{% trans "Home" %}</a> &rsaquo;
+        <a href="{% url 'admin:app_list' app_label=opts.app_label %}">{{ opts.app_config.verbose_name }}</a> &rsaquo;
+        <a href="{% url opts|admin_urlname:'changelist' %}">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
+        <a href="../../">{{ original|truncatewords:"18" }}</a> &rsaquo;
```
Comment (FinalAngel) [inline cms/templates/admin/cms/page/history/revision_header.html]: same here @yakky @mkoistinen

File: cms/templates/admin/cms/page/history/recover_header.html (hunk 1 above)
Comment (yakky) [inline cms/templates/admin/cms/page/history/recover_header.html]: @FinalAngel where this is used? if you have the current object, something like ``{% url opts|admin_urlname:'change' object.pk %}`` should work

**Comment by yakky** [comment]
@FinalAngel  Is this for 3.2.x?

File: cms/templates/admin/cms/page/history/recover_header.html (hunk 1 above)
Comment (FinalAngel) [inline cms/templates/admin/cms/page/history/recover_header.html]: @yakky I'm trying ``{% url opts|admin_urlname:'recover' %}`` but it requires an argument to render ``/recover/id/`` and I  need both cases for:

``/recover/``
``/recover/80/``

File: cms/templates/admin/cms/page/history/revision_header.html [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-    <a href="../../">{{ original|truncatewords:"18" }}</a> &rsaquo;
-    <a href="../">{% trans "History" %}</a> &rsaquo;
-    {% blocktrans with opts.verbose_name as verbose_name %}Revert {{ verbose_name }}{% endblocktrans %}
-</div>
+    <div class="breadcrumbs">
+        <a href="{% url 'admin:index' %}">{% trans "Home" %}</a> &rsaquo;
+        <a href="{% url 'admin:app_list' app_label=opts.app_label %}">{{ opts.app_config.verbose_name }}</a> &rsaquo;
+        <a href="{% url opts|admin_urlname:'changelist' %}">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
+        <a href="{% url opts|admin_urlname:'recoverlist' %}">{{ original|truncatewords:"18" }}</a> &rsaquo;
+        <a href="../">{% trans "History" %}</a> &rsaquo;
```
Comment (FinalAngel) [inline cms/templates/admin/cms/page/history/revision_header.html]: I got it to work for recoverlist @yakky just recover remains. I found the code ``reversion``. Dunno though where this view is used.

**Comment by FinalAngel** [comment]
@yakky we could actually consider getting this into 3.2.x, you wanna backport or should I create a new PR?

**Comment by yakky** [comment]
@FinalAngel I agree it's a good thing for 3.2, asking because it's opened on develop. If this applies cleanly on 3.2 branch I think you can open a PR targeting 3.2.
Backporting is always hard, It's much better forward porting, whenever possible

File: cms/templates/admin/cms/page/history/revision_header.html (hunk 3 above)
Comment (yakky) [inline cms/templates/admin/cms/page/history/revision_header.html]: I'll check this later today

**Pull request closed without merging**